### PR TITLE
1862: Parliament Round; Phases; Train roster; Random start

### PR DESCRIPTION
**Pull request by roseundy**
Parliament Round should be more or less final

### CODE REVIEW

File: lib/engine/game/g_1862/game.rb [hunk 1]
```diff
@@ -297,9 +405,99 @@ def init_companies(players)
...
+            remove_reservation(corp)
+            @companies.delete(@companies.find { |c| c.id == corp.id })
+
+            @log << "Removing #{corp.name} from game"
+          end
+
+          @chartered = {}
+
+          # randomize and distribute train permits
+          permit_list = %i[
```
Comment (jenf) [inline lib/engine/game/g_1862/game.rb:434]: this feels like it can be tidied a bit to generate it.

File: lib/engine/game/g_1862/game.rb [hunk 2]
```diff
@@ -297,9 +405,99 @@ def init_companies(players)
...
+              city.reservations.delete(corporation) if city.reserved_by?(corporation)
+            end
+          end
+        end
+
+        def ipoable_railroads
+          available_railroads.reject(&:ipoed)
+        end
+
+        def available_railroads
```
Comment (jenf) [inline lib/engine/game/g_1862/game.rb:476]: i'm not sure about mixing corporation and railroads in terminology here.

Comment (roseundy) [reply lib/engine/game/g_1862/game.rb:476]: I wasn't either, but I didn't want to have both available_corporations and corporation_available?
I'll see if I can find a less offensive name.

File: lib/engine/game/g_1862/game.rb [hunk 3]
```diff
@@ -297,9 +405,82 @@ def init_companies(players)
...
+          charter = @companies.find { |c| c.id == corporation.id }
+          charter.owner = entity
+          entity.companies << charter
+          @log << "#{entity.name} is under obligation for #{corporation.name}"
+          @chartered[corporation] = true
+        end
+
+        def float_corporation(corporation)
+          super
+          charter = @companies.find { |c| c.id == corporation.id }
```
Comment (jenf) [inline lib/engine/game/g_1862/game.rb:478]: Company_by_id should be equivalent?

**Review by jenf — COMMENTED** [review]
Just the one remaining comment

File: lib/engine/game/g_1862/game.rb (hunk 3 above)
Comment (roseundy) [reply lib/engine/game/g_1862/game.rb:478]: OK.

**Review by jenf — APPROVED** [review]

**Pull request merged**